pkg/apiserver/webhooks/datavolume-validate.go

```diff
@@ -263,50 +281,90 @@ func (wh *dataVolumeValidatingWebhook) validateDataVolumeSpec(request *admission
-			if err != nil {
-				if k8serrors.IsNotFound(err) {
-					causes = append(causes, metav1.StatusCause{
-						Type:    metav1.CauseTypeFieldValueNotFound,
-						Message: fmt.Sprintf("Source PVC %s/%s doesn't exist", spec.Source.PVC.Namespace, spec.Source.PVC.Name),
-						Field:   field.Child("source", "PVC").String(),
-					})
-					return causes
-				}
-			}
-			valid, sourceContentType, targetContentType := validateContentTypes(sourcePVC, spec)
-			if !valid {
-				causes = append(causes, metav1.StatusCause{
-					Type:    metav1.CauseTypeFieldValueInvalid,
-					Message: fmt.Sprintf("Source contentType (%s) and target contentType (%s) do not match", sourceContentType, targetContentType),
-					Field:   field.Child("PVC").String(),
-				})
-				return causes
-			}
-
-			var targetResources v1.ResourceRequirements
-			if spec.PVC != nil {
-				targetResources = spec.PVC.Resources
-			} else {
-				targetResources = spec.Storage.Resources
-			}
-			err = controller.ValidateCloneSize(sourcePVC.Spec.Resources, targetResources)
-			if err != nil {
-				causes = append(causes, metav1.StatusCause{
-					Type:    metav1.CauseTypeFieldValueInvalid,
-					Message: err.Error(),
-					Field:   field.Child("PVC").String(),
-				})
-				return causes
+			cause := wh.validateDataVolumeSourcePVC(spec.Source.PVC, field.Child("source", "PVC"), spec)
+			if cause != nil {
+				causes = append(causes, *cause)
 			}
 		}
 	}
 
 	return causes
 }
 
+func (wh *dataVolumeValidatingWebhook) validateSourceRef(request *admissionv1.AdmissionRequest, spec *cdiv1.DataVolumeSpec, field *k8sfield.Path, namespace *string) *metav1.StatusCause {
+	if spec.SourceRef.Kind == "" {
+		return &metav1.StatusCause{
+			Type:    metav1.CauseTypeFieldValueInvalid,
+			Message: fmt.Sprintf("Missing sourceRef kind"),
+			Field:   field.Child("sourceRef", "Kind").String(),
+		}
+	}
+	if spec.SourceRef.Kind != cdiv1.DataVolumeDataSource {
+		return &metav1.StatusCause{
+			Type:    metav1.CauseTypeFieldValueInvalid,
+			Message: fmt.Sprintf("Unsupported sourceRef kind %s, currently only %s is supported", spec.SourceRef.Kind, cdiv1.DataVolumeDataSource),
+			Field:   field.Child("sourceRef", "Kind").String(),
+		}
+	}
+	if spec.SourceRef.Name == "" {
+		return &metav1.StatusCause{
+			Type:    metav1.CauseTypeFieldValueInvalid,
+			Message: fmt.Sprintf("Missing sourceRef name"),
+			Field:   field.Child("sourceRef", "Name").String(),
+		}
+	}
+	if request.Operation != admissionv1.Create {
+		return nil
+	}
+	ns := namespace
+	if spec.SourceRef.Namespace != nil && *spec.SourceRef.Namespace != "" {
+		ns = spec.SourceRef.Namespace
+	}
+	dataSource, err := wh.cdiClient.CdiV1beta1().DataSources(*ns).Get(context.TODO(), spec.SourceRef.Name, metav1.GetOptions{})
+	if err != nil {
+		return &metav1.StatusCause{
+			Type:    metav1.CauseTypeFieldValueNotFound,
+			Message: fmt.Sprintf("SourceRef %s/%s/%s doesn't exist, %v", spec.SourceRef.Kind, *ns, spec.SourceRef.Name, err),
```

Comment:
Error doesn't necessarily mean the resource does not exist.  Could be a problem communicating with apiserver

Reply:
Not a huge deal, just thinking that if returning `CauseTypeFieldValueNotFound` then explicitly check for not found and return more generic error if something else fails